grammars/puppet.cson

```diff
@@ -394,23 +256,252 @@
+          '1':
+            'name': 'punctuation.parameter.end.puppet'
+        'patterns': [
+          {
+            'include': '#puppet'
+          }
+        ]
+      }
+      {
+        'include': '#variables'
+      }
+      {
+        'include': '#strings'
+      }
+    ]
+  'comparisons':
+    'begin': '(^|\\s)\\s*(if|else|unless|elsif)\\s+'
+    'beginCaptures':
+      '2':
+        'name': 'keyword.control.puppet'
+    'end': '{'
+    'patterns': [
+      {
+        'include': '#parameter-default-types'
+      }
+      {
+        'include': '#comparison-operators'
+      }
+      {
+        'include': '#negation-operators'
+      }
+    ]
+  'comparison-operators':
+    'match': '(?:\\s|^)(={2}|!=|>|<|>=|<=|=~|!~|in)(?:\\s|^)'
+    'name': 'keyword.control.puppet'
+  'negation-operators':
+    'captures':
+      '1':
+        'name': 'negation.keyword.control.puppet'
+    'match': '(?:\\s|^)(not|!)(?:\\s|^)'
+    'name': 'meta.keyword.control.puppet'
+  'numbers':
+    'captures':
+      '1':
+        'name': 'constant.numeric.puppet'
+    'match': '(?:(?:\\D&\\W)|\\s|^|\\[|\\{|,|;|\\()(\\d+([.]\\d+)?)(?!\\w)'
+    'name': 'meta.control.puppet'
+  'define-parameters':
+    'begin': '\\('
+    'beginCaptures':
+      '0':
+        'name': 'punctuation.definition.classparameter.begin.puppet'
+    'end': '\\)'
+    'endCaptures':
+      '0':
+        'name': 'punctuation.definition.classparameter.end.puppet'
+    'name': 'meta.classparameter.language.puppet'
+    'patterns': [
+      {
+        'include': '#parameter-default-types'
+      }
+    ]
+
+  'definitions':
+    'begin': '^\\s*(node|define|class|application)\\s+((?:[-_A-Za-z0-9"\'.]+::)*[-_A-Za-z0-9"\'.]+)\\s*'
+    'beginCaptures':
+      '1':
+        'name': 'storage.type.puppet'
+      '2':
+        'name': 'entity.name.type.class.puppet'
+    'end': '{'
+    'endCaptures':
+      '0':
+        'name': 'punctuation.definition.class.begin.puppet'
+    'name': 'meta.definition.class.puppet'
+    'patterns': [
+      {
+        'begin': '\\b(inherits)\\b\\s+'
+        'beginCaptures':
+          '1':
+            'name': 'storage.modifier.puppet'
+        'end': '{'
+        'endCaptures':
+
+            'name': 'punctuation.definition.class.begin.puppet'
+        'name': 'meta.definition.class.inherits.puppet'
+        'patterns': [
+          {
+            'match': '\\b((?:[-_A-Za-z0-9".]+::)*[-_A-Za-z0-9".]+)\\b'
+            'name': 'entity.other.inherited-class.puppet'
+          }
+        ]
+      }
+      {
+        'include': '#define-parameters'
+      }
+    ]
+  'resources':
+    'begin': '((?:[-_A-Za-z0-9".]+::)*[-_A-Za-z0-9".]+)\\s*{'
+    'beginCaptures':
+      '1':
+        'name': 'storage.type.puppet'
+    'name': 'meta.definition.resource.puppet'
+    'end': '}'
+    'patterns': [
+      {
+        'match': ':'
+        'captures':
+          '0':
+            'name': 'punctuation.classtitle.puppet'
+        'name': 'meta.title.puppet'
```

Comment:
I don't think this is needed.  `meta` groups generally are used to match a wide section of code for identification purposes and not highlighting purposes.